mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx

```diff
@@ -0,0 +1,303 @@
+  function findNodesFromTree(
+    tree: any,
+    itemsToFind: string[],
+    results: RenderTree[],
+  ) {
+    tree.forEach((item: RenderTree) => {
+      if (itemsToFind.includes(item.id)) {
+        results.push(item);
+      } else {
+        if (item.children && item.children.length > 0) {
+          return findNodesFromTree(item.children, itemsToFind, results);
+        }
+      }
+    });
+    return results;
+  }
+
+  function clearTreeSearch() {
+    setTreeSelections([]);
+    setTreeData(cloneDeep(treeDataOriginal));
+    setExpanded();
+  }
+
+  function doFiltering(
+    tree: RenderTree[],
+    nameToFind: string,
+    results: { nodeIds: string[]; childNodeIds: string[] },
+  ) {
+    tree.forEach((item) => {
+      if (
+        item.name &&
+        item.name.toLowerCase().includes(nameToFind.toLowerCase())
+      ) {
+        results.nodeIds.push(item.id);
+        if (item.children && item.children.length > 0) {
+          item.children.forEach((child) => {
+            results.childNodeIds.push(child.id);
+          });
+        }
+      }
+      if (item.children && item.children.length > 0) {
+        return doFiltering(item.children, nameToFind, results);
+      }
+    });
+    return results;
+  }
+
+  function getElementPathsFromTree(
+    treeData: RenderTree[],
+    nodeIds: string[],
+    results: string[],
+  ) {
+    treeData.forEach((item) => {
+      if (nodeIds.includes(item.id)) {
+        if (item.elementPath != null) {
+          results.push(item.elementPath);
+        }
+      }
+      if (item.children && item.children.length > 0) {
+        return getElementPathsFromTree(item.children, nodeIds, results);
+      }
+    });
+    return results;
+  }
+
+  // Used by tree select and filtering
+  function getAllNodeIdsOnPathToLeaf(nodeIds: string[]) {
+    const elementPaths = getElementPathsFromTree(treeData, nodeIds, []);
+    const nodesToSelect: Set<string> = new Set();
+
+    elementPaths.forEach((path) => {
+      const nodes = path.split('.');
+      nodes.forEach((node) => {
+        nodesToSelect.add(node);
+      });
+    });
+    return Array.from(nodesToSelect);
+  }
+
+  const handleExpandClick = () => {
+    const allTreeNodes: string[] = [];
+    treeData.forEach(({ children, id }) => {
+      if (children && children?.length > 0) {
+        allTreeNodes.push(id.toString());
+      }
+    });
+    setTreeExpanded((oldExpanded) => {
+      return oldExpanded.length === 0 ? allTreeNodes : [];
+    });
+  };
+
+  function expandAndSelectNodes(nodeIds: string[]) {
+    if (nodeIds.length > 0) {
+      const nodeIdsToExpand = getAllNodeIdsOnPathToLeaf(nodeIds);
+      setTreeExpanded(nodeIdsToExpand);
+      setTreeSelections(nodeIds);
+    }
+  }
+
+  function searchFromTree(input: any) {
+    clearTreeSearch();
+    const hits = { nodeIds: [], childNodeIds: [] };
+    doFiltering(treeData, input.toString(), hits);
+    expandAndSelectNodes(hits.nodeIds);
+  }
+
+  function handleTreeClick(
```

Comment:
I have trouble understanding the need for the handleTreeClick and handleTreeToggle as intermediary functions here. Could you not put setTreeSelections(nodeIds) and setTreeExpanded(nodeIds) in their place for more concise code? And I'm curious why is the unused argument (event) typed when precise typing is missing in many other places. I'd like to learn the reasoning behind it.